modules/kubernetes/src/test/java/org/apache/ignite/kubernetes/discovery/TestClusterClientConnection.java

```diff
@@ -49,4 +49,44 @@ public void testClientConnectsToCluster() throws Exception {
         cache.put(1, 2);
         assertEquals(2, cache.get(1));
     }
+
+    @Test
+    public void testClientReConnectsToClusterAfterPodIpChange() throws Exception {
+        mockServerResponse();
+
+        IgniteConfiguration cfg = getConfiguration(getTestIgniteInstanceName(), false);
+
+        IgniteEx crd = startGrid(cfg);
+        String crdAddr = crd.localNode().addresses().iterator().next();
+
+        mockServerResponse(crdAddr);
+
+        ClientConfiguration ccfg = new ClientConfiguration();
+        ccfg.setAddressesFinder(new ThinClientKubernetesAddressFinder(prepareConfiguration()));
+        IgniteClient client = Ignition.startClient(ccfg);
+
+        ClientCache cache = client.createCache("cache");
+        cache.put(1, 2);
+        assertEquals(2, cache.get(1));
+
+        //stop node and change port still can connect
+        Ignition.stop(crd.name(), true);
+        int newPort = 10801;
+        ccfg.setAddressesFinder(new ThinClientKubernetesAddressFinder(prepareConfiguration(),newPort));
+        mockServerResponse(5,crdAddr);
+
+        cfg = getConfiguration(getTestIgniteInstanceName(), false);
+        cfg.setSqlConnectorConfiguration(new SqlConnectorConfiguration().setPort(newPort));
```

Comment:
`setSqlConnectorConfiguration` is deprecated, please use `setClientConnectorConfiguration` instead.

Reply:
> `setSqlConnectorConfiguration` is deprecated, please use `setClientConnectorConfiguration` instead.

i found ClientListenerProcessor method isNotDefault(cliConnCfg)  will check ClientConnectorConfiguration isDefault then it will return false，then   cliConnCfg = new ClientConnectorConfiguration();
so i set newPort will not take effect .then i use SqlConnectorConfiguration to set newPort. I don't know other method to set a newPort

Reply:
> `setSqlConnectorConfiguration` is deprecated, please use `setClientConnectorConfiguration` instead.
i'm sorry. when i use setClientConnectorConfiguration  to set a newPort then startGrid(cfg), i found ClientListenerProcessor method isNotDefault(cliConnCfg)  will check ClientConnectorConfiguration isDefault then it will return false，then   cliConnCfg = new ClientConnectorConfiguration();
   so i set newPort will not take effect .then i use SqlConnectorConfiguration to set newPort. I don't know other method to set a newPort

Reply:
Not sure I understand the problem. Replacing `setSqlConnectorConfiguration` with `setClientConnectorConfiguration` seems to work, the test passes. I've pushed the change to you branch, please review.

Reply:
> Not sure I understand the problem. Replacing `setSqlConnectorConfiguration` with `setClientConnectorConfiguration` seems to work, the test passes. I've pushed the change to you branch, please review.
thank you reply。i debug cfg.getClientConnectorConfiguration() ，and type is ClientConnectorConfigurationEx，and i use  cfg.getClientConnectorConfiguration().setPort() so it not success。
thank you